server/tserver/src/main/java/org/apache/accumulo/tserver/tablet/Tablet.java

```diff
@@ -1412,12 +1412,40 @@ private void closeConsistencyCheck() {
 
     if (!otherLogs.isEmpty() || !currentLogs.isEmpty() || !referencedLogs.isEmpty()) {
       String msg = "Closed tablet " + extent + " has walog entries in memory currentLogs = "
-          + currentLogs + "  otherLogs = " + otherLogs + " refererncedLogs = " + referencedLogs;
+          + currentLogs + "  otherLogs = " + otherLogs + " referencedLogs = " + referencedLogs;
       log.error(msg);
       throw new RuntimeException(msg);
     }
 
-    // TODO check lastFlushID and lostCompactID - ACCUMULO-1290
+    // If a table hasn't been flushed before it was closed then lastFlushID will be -1 while
+    // getFlushID will be 0.
+    try {
+      long flushID = getFlushID();
```

Comment:
this will read the current table wide flush ID from ZK.  That is not what we want to check here.  Instead, we want to check if lastFlushId matches what is stored in the metdata table for this tablet.  The lastFlushId instance var should contain whats in the metadata table and it should be updated in memory when the metadata table is updated.  We can change the Ample code that reads the tablet metadata to include flush id and then do something like the following.

```java
if(tabletMeta.getFlushId().orElse(-1) != lastFlushId) {
  // mismatch with tablet metadata , so throw exception
}
```

Not sure if the case where flushId was never set it correct in the code above.  Need to investigate further.

Reply:
So I was looking into this and I have some questions.

You mentioned that we can change the Ample code that reads the tablet metadata to include flushID and do what you had suggested. I was curious what part of code that may be. I could not identify it, but I did see that Ample already had knowledge of flushID in general. 

Would the change just deal with how the variable is updated? Or would it be dealing with something more complex?

Reply:
>  I was curious what part of code that may be.

There is some code like the following earlier in the method that uses Ample to read the tablets metadata. Can add the FLUSH_ID to that code.

```java
      var tabletMeta = context.getAmple().readTablet(extent, ColumnType.FILES, ColumnType.LOGS,
          ColumnType.ECOMP, ColumnType.PREV_ROW, ColumnType.FLUSH_ID);
```

Reply:
Thank you, that is very helpful! 

Would this be a similar process for COMPACT_ID?

Reply:
> Would this be a similar process for COMPACT_ID?

Yeah I think so.